packages/router/src/grpc/view-protocol-server/transaction-planner.ts

```diff
@@ -24,6 +26,11 @@ export const transactionPlanner: Impl['transactionPlanner'] = async (req, ctx) =
     fmdParams,
   });
 
+  const fetchedGasPrices = await gasPrices(req, ctx);
+  if (fetchedGasPrices.gasPrices) {
+    planner.setGasPrices(new GasPrices(fetchedGasPrices.gasPrices));
+  }
```

Comment:
At the moment, these lines cause the following error when I press "Send":
![image](https://github.com/penumbra-zone/web/assets/1121544/8b6fe88c-3d40-43a9-b394-ad659d3888b9)

Because of that, we have to wait until @Valentine1898 's work on #203 is finished before we can merge this PR.

Reply:
would like to see this resolved first, yes

Reply:
has this resolved?

Reply:
Yep!

I just tested it again though, and noticed that I'm still not seeing fees in the approval window:

<img width="959" alt="image" src="https://github.com/penumbra-zone/web/assets/1121544/be033403-f2e0-41e8-bec7-80914389b516">

This _may_ be because the amounts are all `0n`, but I'd still expect to see something in the JSON view... @Valentine1898 , since you just worked on it, do you happen to know if the fee amount is expected to be empty when each dimension of the fee is `0n`?

Reply:
https://github.com/penumbra-zone/penumbra/blob/28ee5f71ba620e77b9ddf6ebcd740ed6a5be359c/crates/core/component/fee/src/gas.rs#L67
Based on this formula, if all gas prices are `0n`, then we will get a `0n` gas cost transaction. 
But it is strange that we get the display 
`"amount": {}`
I would expect `"amount": {"lo": 0n, "hi: 0n} `
but maybe it's just a serialization case.

Reply:
@Valentine1898 that is indeed strange. I logged [the `tx` object](https://github.com/penumbra-zone/web/blob/jessepinho/web-308-gas-fees-in-web/apps/extension/src/routes/popup/approval/transaction.tsx#L7-L8) that the `TransactionApproval` component passes to the JSON Viewer, and `amount` is indeed an empty object. I'm looking into this more, but if you have any insight on why that could be even though [I called `setGasPrices`](https://github.com/penumbra-zone/web/blob/jessepinho/web-308-gas-fees-in-web/packages/router/src/grpc/view-protocol-server/transaction-planner.ts#L29-L32) from inside `transactionPlanner`, it'd be great to hear your thoughts!

<img width="1350" alt="image" src="https://github.com/penumbra-zone/web/assets/1121544/086cd00d-c0b4-4c71-a7f3-3241855656e4">

Reply:
Yes, I changed the gas prices values in indexed-db and it didn't affect anything. 
The issue seems to be in this logic 
https://github.com/penumbra-zone/penumbra/blob/28ee5f71ba620e77b9ddf6ebcd740ed6a5be359c/crates/wasm/src/planner.rs#L504

Reply:
The issue seems to be that we're just not passing the fee to planner

Reply:
No, I was wrong, I'm sorry. 
I only changed the blockSpacePrice in indexed-db so it had no effect. After I changed the executionPrice I got the correct display 
<img width="561" alt="image" src="https://github.com/penumbra-zone/web/assets/25391690/5c2e14a8-bc47-4cb6-b6df-6388742c1a24">

Reply:
I still think the mapping of `"amount": {} `is a serialization case, but not a serialization of `ts object -> json`, but possibly a serialization of `wasm -> ts object`